walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,151 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  // Return the possible times at which the meeting be set on the day.
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // All the possible times to arrange a meeting on the day.
+    List<TimeRange> possibleTimes = new ArrayList<>();
+
+    // All the people who are required to attend the meeting. 
+    Collection<String> requestAttendees = request.getAttendees();
+
+    // All the people whose attendance is optional. 
```

Comment:
nit: where practical, it's better for the code to be self-documenting without needing comments -- reserving comments for situations where something non-obvious is happening. In cases like this, it's clear what the variable represents from its name.

Similar nit above -- 'requestAttendees' could be renamed 'requiredAttendees' to remove the need for the comment?